README.md

```diff
@@ -10,33 +10,74 @@
 dotnet add package KubernetesClient
 ```
 
-# Generating the Client Code
+# Usage
 
-## Prerequisites
+## Authentication/Configuration
+You should be able to use an standard KubeConfig file with this library,
+see the `BuildConfigFromConfigFile` function below. Most authentication
+methods are currently supported, but a few are not, see the 
+[known-issues](https://github.com/kubernetes-client/csharp#known-issues)
 
-You'll need a Linux machine with Docker.
+You should also be able to authenticate using the in-cluster service
+account using the `InClusterConfig` function shown below.
 
-The generated code works on all platforms supported by .NET or .NET Core.
+## Sample Code
 
-Check out the generator project into some other directory
-(henceforth `$GEN_DIR`)
+### Creating the client
+```c#
+// Load from the default kubeconfig on the machine.
+var config = KubernetesClientConfiguration.BuildConfigFromConfigFile();
 
-```bash
-cd $GEN_DIR/..
-git clone https://github.com/kubernetes-client/gen
+// Load from a specific file:
+var config = KubernetesClientConfiguration.BuildConfigFromConfigFile(Environment.GetEnvironmentVariable("KUBECONFIG"));
+
+// Load from in-cluster configuration:
+var config = KubernetesClientConfiguration.InClusterConfig()
+
+// Use the config object to create a client.
+var client = new Kubernetes(config);
 ```
 
-## Generating code
+### Listing Objects
+```c#
+var namespaces = client.ListNamespace();
+foreach (var ns in namespaces.Items) {
+    Console.WriteLine(ns.Metadata.Name);
+    var list = client.ListNamespacedPod(ns.Metadata.Name);
+    foreach (var item in list.Items)
+    {
+        Console.WriteLine(item.Metadata.Name);
+    }
+}
+```
 
-```bash
-# Where REPO_DIR points to the root of the csharp repository
-cd ${REPO_DIR}/csharp/src/KubernetesClient
-${GEN_DIR}/openapi/csharp.sh generated ../csharp.settings
+### Creating and Deleting Objects
+```c#
+var ns = new V1Namespace
+{
+    Metadata = new V1ObjectMeta
+    {
+        Name = "test"
+    }
+};
+
+var result = client.CreateNamespace(ns);
+Console.WriteLine(result);
+
+var result = client.CreateNamespace(ns);
+Console.WriteLine(result);
+
+var status = client.DeleteNamespace(ns.Metadata.Name, new V1DeleteOptions());
 ```
 
-# Usage
+### Deleting Objects
```

Comment:
Covered under `Creating and Deleting Objects` or did you mean to move the delete code from there to under here?

Reply:
oops. fixed.